### PR TITLE
Use actions `v2`

### PR DESCRIPTION
**Pull request by claudiu725**
- use `v2` of `contracts.yml` from `mx-sc-actions`
  - this uses `mxpy` instead of `erdpy`, which provides `run-scenarios` instead of `mandos-test`
- updated `multiversx_sc_scenario::run_go` to use `run-scenarios`
  - breaking change note: this means that existing github actions using `contracts.yml@v1` will break and have to be updated to use `v2`
- added handling for the exit status of several more processes, as contract builds were silently failing but the status was reported as successful
- renamed and updated scripts with the following changes:
  - from `elrond.json` to `multiversx.json`
  - from `elrond.workspace.json` to `multiversx.workspace.json`
- renamed some variables

### CODE REVIEW

**Review by dorin-iancu — DISMISSED** [review]

**Review by ovstinga — APPROVED** [review]

File: framework/scenario/src/scenario_go_runner.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ pub fn run_go<P: AsRef<Path>>(relative_path: P) {
     let mut absolute_path = std::env::current_dir().unwrap();
     absolute_path.push(relative_path);
 
-    let output = Command::new("mandos-test")
+    let output = Command::new("run-scenarios")
```
Comment (andrei-marinica) [inline framework/scenario/src/scenario_go_runner.rs:13]: Let's make a separate function with this flag, run_go_v2, for backwards compatibility.

I will refactor the whole thing early next week.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: framework/scenario/src/scenario_go_runner.rs (hunk 1 above)
Comment (andrei-marinica) [reply framework/scenario/src/scenario_go_runner.rs:13]: I will also provide a proper test generator for the files where this is called.

Comment (andrei-marinica) [reply framework/scenario/src/scenario_go_runner.rs:13]: Or, another option: check that the binary exists before calling, and fallback to the old one if not.

Comment (claudiu725) [reply framework/scenario/src/scenario_go_runner.rs]: I added a fallback for this, which tries to run `run-scenarios` and if that fails with a `not found` error, it tries the old binary.

**Review by andrei-marinica — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by ovstinga — APPROVED** [review]

**Pull request merged**